api/src/opentrons/protocol_engine/commands/calibration/calibrate_pipette.py

```
@@ -0,0 +1,82 @@
+"""Calibrate-pipette command for OT3 hardware. request, result, and implementation models."""
+from __future__ import annotations
+
+from typing import Optional, Type, List
+from typing_extensions import Literal
+from pydantic import BaseModel, Field
+
+from ..command import (
+    AbstractCommandImpl,
+    BaseCommand,
+    BaseCommandCreate,
+)
+from ...errors import ErrorOccurrence
+from ...validation import ensure_ot3_hardware
+
+from opentrons.hardware_control import HardwareControlAPI
+from opentrons.hardware_control import ot3_calibration as calibration
+from opentrons.hardware_control.types import OT3Mount
+from opentrons.types import Point
+
+CalibratePipetteCommandType = Literal["calibratePipette"]
+
+
+class CalibratePipetteParams(BaseModel):
+    """Payload required to calibrate-pipette."""
+
+    mount: OT3Mount = Field(..., description="Instrument mount to calibrate.")
+
+
+class CalibratePipetteResult(BaseModel):
+    """Result data from the execution of a calibrate-pipette command."""
+
+    pipetteOffset: Point = Field(
+        ..., description="Offset of calibrated pipette."
+    )
+    errors: Optional[List[ErrorOccurrence]] = Field(
```

this feels redundant? we can get the errors from the StateSummery errors property or we let the errors bubble up

---

I think we definitely don't want errors bubbling up directly from the hardware controller if we can avoid it b/c I was under the impression that the App has much nicer ways to handle errors returned from protocol engine. @b-cooper or @shlokamin, please correct me if I'm wrong here.

What types of errors does the `StateSummary` catch normally?

---

It is redundant to have an `errors: Optional[List[ErrorOccurrence]]` field. It's nearly exactly what a failed command is, with its `.error` field.

I think we need to identify the particular goals of error reporting here.

* Is there a specific, bounded set of conditions that a client should react and adapt to? Like: "a pipette isn't attached; attach one and try again"?
    * If so, we need to enumerate those conditions more specifically than the generic `ErrorOccurrence` type.
    * We can choose how to report those conditions (here, or via a failed command with an `.error`) based on what those conditions are, specifically. 
* Or do we just want this as a catch-all for "something unexpected happened, so the app should show some kind of error indicator to the user"?
    * If so, we should just use `.error` for this (instead of `params.errors` as currently written).

---

We agreed upon removing the error property from the command result. The UI will check if the command was successful. If not it will check the error field.